hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -235,7 +235,13 @@ object Simplify {
 
       case TableMapGlobals(child, Ref("global", _)) => child
 
-      case MatrixMapRows(child, Ref("va", _), None) => child
+      case MatrixMapRows(child, Ref("va", _)) => child
+
+      case t@MatrixKeyRowsBy(MatrixKeyRowsBy(child, _, _), keys, false) if canRepartition(t) =>
```

Comment:
What does the `canRepartition` check for?

Reply:
@catoverdrive added that to prevent the simplifier from changing partitioning in cases where that could affect the result of random number generation. Basically we have to guard any simplification rule that changes partitioning.